### PR TITLE
Recorder extension improvements

### PR DESCRIPTION
**Pull request by OneEyedSpaceFish**
See #5177

### CODE REVIEW

File: ext/Recorder.php [hunk 1]
```diff
@@ -434,4 +560,38 @@ protected function isStepIgnored($step)
...
+    }
+
+    /**
+     * @param string $message
+     */
+    protected function writeln($message)
+    {
+        parent::writeln(
+            $this->ansi
+                ? $message
```
[the remark this answers is not in the document]
Comment (OneEyedSpaceFish) [reply ext/Recorder.php]: If you guys agree with Mr Nitpick (:smiley:) I'll 'fix' this

[the remark this answers is not in the document]
Comment (OneEyedSpaceFish) [reply ext/Recorder.php]: Although I'm not sure if this is correct. This is a multi-line ternary operator.

**Comment by OneEyedSpaceFish** [comment]
> Could you make screenshot/gif/video of how recorder looks now?

Sure thing! I'll create a demo test soon-ish.
While I'm doing that, there's a screenshot added in #5177 for the `records.html` file:
https://github.com/Codeception/Codeception/issues/5177#issue-361696318
That's still up to date, as there were no visual changes to that since I've uploaded that.
I'll create a video that includes both `records.html` and `index.html`.

On another note;
As you'd probably know the answer to this; is there somewhere we'd like to document the new changes / behavior / extra features (other than as comments in `Recorder.php`)?

FYI, I tried to keep any new features in a turned off state by default, but if you'd like, those could be enabled as well. Just let me know.

**Comment by OneEyedSpaceFish** [comment]
@DavertMik 
MP4: https://drive.google.com/file/d/1IhwAS7fgBYsKXPPIjRQ1X_CmRijBWg4t/view?usp=sharing
GIF: https://drive.google.com/file/d/1Q6zxAlmclF1qpbmfHE7g63d3jfV78s32/view?usp=sharing

**Comment by DavertMik** [comment]
> As you'd probably know the answer to this; is there somewhere we'd like to document the new changes / behavior / extra features (other than as comments in Recorder.php)?

Nope, everything should go to comments. They will be published on the site in Extensions section.
If you think there is something to add into Guides, you can update 03-Acceptance Testing chapter

**Comment by OneEyedSpaceFish** [comment]
@DavertMik there's a 4 line change that would need to be done still unfortunately.
This scenario is also not being handled in previous versions of recordings so it should be fine as the change is not introducing any new bugs or issues, but in production this morning I noticed that if the webDriver fails to hydrate inside the WebDriverModule we are still encountering an exception.

If this is an issue feel free to skip this from being included in the next release, however, I'd like this scenario being handled, I understand that this is pretty much a doomsday scenario, but if this occurs the exception obscures the actual exception message / failure in the tests report.

If I'd make that change quickly (4-10 lines tops), would you be happy to merge that before the new release happens?

**Pull request merged**